internal/utils/gke.go

```diff
@@ -0,0 +1,508 @@
+package utils
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	gkev1 "github.com/rancher/gke-operator/pkg/apis/gke.cattle.io/v1"
+	wranglerv1 "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	gkeapi "google.golang.org/api/container/v1"
+	"google.golang.org/api/option"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// Node Pool Status
+const (
+	// NodePoolStatusUnspecified - Not set.
+	NodePoolStatusUnspecified = "STATUS_UNSPECIFIED"
+	// NodePoolStatusProvisioning The PROVISIONING state indicates the node pool is
+	// being created.
+	NodePoolStatusProvisioning = "PROVISIONING"
+	// NodePoolStatusRunning The RUNNING state indicates the node pool has been
+	// created
+	// and is fully usable.
+	NodePoolStatusRunning = "RUNNING"
+	// NodePoolStatusRunningWithError The RUNNING_WITH_ERROR state indicates the
+	// node pool has been created
+	// and is partially usable. Some error state has occurred and
+	// some
+	// functionality may be impaired. Customer may need to reissue a
+	// request
+	// or trigger a new update.
+	NodePoolStatusRunningWithError = "RUNNING_WITH_ERROR"
+	// NodePoolStatusReconciling The RECONCILING state indicates that some work is
+	// actively being done on
+	// the node pool, such as upgrading node software. Details can
+	// be found in the `statusMessage` field.
+	NodePoolStatusReconciling = "RECONCILING"
+	//  NodePoolStatusStopping The STOPPING state indicates the node pool is being
+	// deleted.
+	NodePoolStatusStopping = "STOPPING"
+	// NodePoolStatusError The ERROR state indicates the node pool may be unusable.
+	// Details
+	NodePoolStatusError = "ERROR"
+)
+
+// Network Providers
+const (
+	// NetworkProviderUnspecified is an unspecified netowrk provider
+	NetworkProviderUnspecified = "PROVIDER_UNSPECIFIED"
+
+	// NetworkProviderCalico describes the calico provider
+	NetworkProviderCalico = "CALICO"
+)
+
+// Cluster Status
+const (
+	// ClusterStatusProvisioning The PROVISIONING state indicates the cluster is
+	// being created.
+	ClusterStatusProvisioning = "PROVISIONING"
+
+	// ClusterStatusRunning The RUNNING state indicates the cluster has been
+	// created and is fully
+	ClusterStatusRunning = "RUNNING"
+
+	// ClusterStatusStopping The STOPPING state indicates the cluster is being
+	// deleted.
+	ClusterStatusStopping = "STOPPING"
+
+	// ClusterStatusError is a ClusterStatus enum value
+	ClusterStatusError = "ERROR"
+
+	// ClusterStatusReconciling The RECONCILING state indicates that some work is
+	// actively being done on
+	ClusterStatusReconciling = "RECONCILING"
+
+	// ClusterStatusUnspecified - Not set.
+	ClusterStatusUnspecified = "STATUS_UNSPECIFIED"
+
+	// ClusterStatusDegraded The DEGRADED state indicates the cluster requires user
+	// action to restore
+	ClusterStatusDegraded = "DEGRADED"
```

Comment:
Are these available from a gke package? If not, this is good. I appreciate the comments.

Reply:
The gke package doesn't provide these as consts, they're just documented strings.